### PR TITLE
Fix/14578 no internet prevents clicking on srs

### PR DESCRIPTION
**Pull request by puneetmahali1**
## Description
No internet connections prevents clicking on Self test submission

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-14578

## Screenshots
https://user-images.githubusercontent.com/72390277/214326263-249aa7be-f80d-4239-b4e2-1dd7447175d0.MP4

### CODE REVIEW

**Comment by Ein-Tim** [comment]
So IIUC when you are not connected to the internet and click on the tile nothing happens.  Like nothing at all? I don't think that's a good idea! Some sort of error should be shown to the user, otherwise this could lead to very confused users.

**Comment by thomasaugsten** [comment]
@Ein-Tim This is under certain circumstances the situation with 3.0 and this PR will fix the issue.

**Comment by Ein-Tim** [comment]
@thomasaugsten So there will actually be some sort of UI information presented to the user why this doesn't work?

**Comment by thomasaugsten** [comment]
Depends if he never has an internet connection he will receive the error that we cannot check his phone time.
If its a temporary no internet he will receive an no internet error when he tries to submit

**Review by 30mar — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Services/Risk/Provider/Helper/CachedAppConfiguration.swift [hunk 1]
```diff
@@ -31,9 +31,10 @@ final class CachedAppConfiguration: AppConfigurationProviding {
 	// MARK: Protocol AppConfigurationProviding
 
 	func appConfiguration(forceFetch: Bool = false) -> AnyPublisher<SAP_Internal_V2_ApplicationConfigurationIOS, Never> {
+		let isOnline = InternetConnectivityMonitor.shared.isDeviceOnline
```
Comment (30mar) [inline src/xcode/ENA/ENA/Source/Services/Risk/Provider/Helper/CachedAppConfiguration.swift:34]: aren't we doing a shared class now as discussed?

**Review by 30mar — CHANGES_REQUESTED** [review]
aren't we doing a shared class for checking the network now as discussed?

File: src/xcode/ENA/ENA/Source/Services/Risk/Provider/Helper/CachedAppConfiguration.swift (hunk 1 above)
Comment (puneetmahali1) [reply src/xcode/ENA/ENA/Source/Services/Risk/Provider/Helper/CachedAppConfiguration.swift:34]: InternetConnectivityMonitor is the shared class which I got from Felix after merged his reopened PR.

**Review by 30mar — APPROVED** [review]

**Review by naveeddotio — APPROVED** [review]

**Pull request merged**